src/tyr/isochrone_serializer.cc

```diff
@@ -18,71 +18,80 @@ using rgba_t = std::tuple<float, float, float>;
 namespace valhalla {
 namespace tyr {
 
-template <class coord_t>
 std::string
 serializeIsochrones(const Api& request,
-                    const typename midgard::GriddedData<coord_t>::contours_t& grid_contours,
+                    const std::map<const midgard::IsoMetrics, const contours_t>& grid_contours,
                     bool polygons,
-                    const std::unordered_map<float, std::string>& colors,
+                    const std::unordered_map<const midgard::IsoMetrics,
+                                             std::unordered_map<float, std::string>>& colors,
                     bool show_locations) {
   // for each contour interval
   int i = 0;
   auto features = array({});
-  for (const auto& interval : grid_contours) {
-    auto color_itr = colors.find(interval.first);
-    // color was supplied
-    std::stringstream hex;
-    if (color_itr != colors.end() && !color_itr->second.empty()) {
-      hex << "#" << color_itr->second;
-    } // or we computed it..
-    else {
-      auto h = i * (150.f / grid_contours.size());
-      auto c = .5f;
-      auto x = c * (1 - std::abs(std::fmod(h / 60.f, 2.f) - 1));
-      auto m = .25f;
-      rgba_t color = h < 60 ? rgba_t{m + c, m + x, m}
-                            : (h < 120 ? rgba_t{m + x, m + c, m} : rgba_t{m, m + c, m + x});
-      hex << "#" << std::hex << static_cast<int>(std::get<0>(color) * 255 + .5f) << std::hex
-          << static_cast<int>(std::get<1>(color) * 255 + .5f) << std::hex
-          << static_cast<int>(std::get<2>(color) * 255 + .5f);
-    }
-    ++i;
+  for (const auto& metric_contours : grid_contours) {
```

Comment:
Looks like a huge change, but really it's just adding an extra loop over the metrics and setting a few more high-level variables.